### PR TITLE
Adding a description of the input of the CNN layer

### PR DESCRIPTION
**Pull request by InterTriplete2010**
Documentation of the input of the CNN layer

### CODE REVIEW

File: cnn_documentation.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Documentation of the layer regarding input weight and height to clarify how the input matrix is treated, especially if it is rectangular.
```
Comment (zoq) [inline cnn_documentation.txt:1]: Adding an extra file to the root directory looks strange to me, do you think you can add the details right after https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/convolution.hpp#L32

That comes with the main advantage that everything we have there is going to be picked up by doxygen the engine we use to generate the documentation.

Comment (InterTriplete2010) [reply cnn_documentation.txt:1]: Just to make sure I am doing the right thing, do you want me to copy the description of the input layer of the CNN in the convolution.hpp file? Is that correct?

Comment (zoq) [reply cnn_documentation.txt:1]: That is correct.

Comment (InterTriplete2010) [reply cnn_documentation.txt:1]: Ok. Done

**Comment by zoq** [comment]
Closing in favour of https://github.com/mlpack/mlpack/pull/2831.

**Pull request closed without merging**